modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteScrollableHitSource.java

```diff
@@ -71,10 +68,6 @@
 import static org.elasticsearch.index.reindex.remote.RemoteResponseParsers.RESPONSE_PARSER;
 
 public class RemoteScrollableHitSource extends ScrollableHitSource {
-    /**
-     * The maximum size of the remote response to buffer. 200mb because bulks beyond 40mb tend to be slow anyway but 200mb is simply huge.
-     */
-    private static final ByteSizeValue BUFFER_LIMIT = new ByteSizeValue(200, ByteSizeUnit.MB);
```

Comment:
I found #21185 which was the reason for the 200MB so this part of the change is not ok. Not sure whether we should increase the default value. After all we want people to use our Rest client for reindexing, and they should not need to provide their own HttpAsyncResponseConsumerFactory in order to do that. Maybe make the default 100MB instead?